linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -1971,6 +1972,26 @@ private void closeInner() {
       final Function2<TSource, TInner, TResult> resultSelector,
       boolean generateNullsOnLeft,
       boolean generateNullsOnRight) {
+    return mergeJoin(outer, inner, outerKeySelector, innerKeySelector, null,
+        resultSelector, generateNullsOnLeft, generateNullsOnRight);
+  }
+
+  /**
+   * Joins two inputs that are sorted on the key, with an extra predicate for non equi-join
+   * conditions (in case of equi-join, it will be null).
+   * Inputs must sorted in ascending order, nulls last.
+   * NOTE: The current API is experimental and subject to change without notice.
+   */
+  @Experimental
```

Comment:
I would suggest `@API(...)`

Reply:
Ok, I'll change it.

Reply:
Unfortunately, when I try to import `org.apiguardian.api.API;` and use  `@API(since = "1.23", status = API.Status.EXPERIMENTAL)` in EnumerableDefaults (which belongs to linq4j), I get a compilation error. Is it possible to use that library there? Do I need to change gradle config files to do so?

Reply:
It comes from `apiguardian`, so you need to add the relevant dependency to `linq4j`: https://github.com/apache/calcite/blob/d234626227954eefffe49f42abec65c649ffe3a6/core/build.gradle.kts#L44

Reply:
added apiguardian